HLTrigger/special/plugins/HLTPixelThrustFilter.cc

```diff
@@ -0,0 +1,107 @@
+#include "RecoLocalTracker/ClusterParameterEstimator/interface/PixelClusterParameterEstimator.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "RecoLocalTracker/Records/interface/TkPixelCPERecord.h"
+#include "PhysicsTools/CandUtils/interface/Thrust.h"
+
+//
+// class declaration
+//
+
+class HLTPixelThrustFilter : public HLTFilter {
+public:
+  explicit HLTPixelThrustFilter(const edm::ParameterSet&);
+  ~HLTPixelThrustFilter() override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  bool hltFilter(edm::Event&,
+                 const edm::EventSetup&,
+                 trigger::TriggerFilterObjectWithRefs& filterproduct) const override;
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> const trackerGeometryRcdToken_;
+  edm::ESGetToken<PixelClusterParameterEstimator, TkPixelCPERecord> const pixelCPERcdToken_;
+  edm::InputTag inputTag_;  // input tag identifying product containing pixel clusters
+  double min_thrust_;       // minimum thrust
+  double max_thrust_;       // maximum thrust
+  edm::EDGetTokenT<edmNew::DetSetVector<SiPixelCluster> > inputToken_;
+};
+
+//
+// constructors and destructor
+//
+
+HLTPixelThrustFilter::HLTPixelThrustFilter(const edm::ParameterSet& config)
+    : HLTFilter(config),
+      trackerGeometryRcdToken_(esConsumes()),
+      pixelCPERcdToken_(esConsumes<PixelClusterParameterEstimator, TkPixelCPERecord>(
+          config.getParameter<edm::ESInputTag>("pixelCPE"))),
+      inputTag_(config.getParameter<edm::InputTag>("inputTag")),
+      min_thrust_(config.getParameter<double>("minThrust")),
+      max_thrust_(config.getParameter<double>("maxThrust")) {
+  inputToken_ = consumes<edmNew::DetSetVector<SiPixelCluster> >(inputTag_);
+}
+
+HLTPixelThrustFilter::~HLTPixelThrustFilter() = default;
+
+void HLTPixelThrustFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  makeHLTFilterDescription(desc);
+  desc.add<edm::InputTag>("inputTag", edm::InputTag("hltSiPixelClusters"));
+  desc.add<edm::ESInputTag>("pixelCPE", edm::ESInputTag{"", "hltESPPixelCPEGeneric"});
+  desc.add<double>("minThrust", 0);
+  desc.add<double>("maxThrust", 0);
+  descriptions.add("hltPixelThrustFilter", desc);
+}
+
+//
+// member functions
+//
+// ------------ method called to produce the data  ------------
+bool HLTPixelThrustFilter::hltFilter(edm::Event& event,
+                                     const edm::EventSetup& iSetup,
+                                     trigger::TriggerFilterObjectWithRefs& filterproduct) const {
+  // All HLT filters must create and fill an HLT filter object,
+  // recording any reconstructed physics objects satisfying (or not)
+  // this HLT filter, and place it in the Event.
+
+  // The filter object
+  if (saveTags())
+    filterproduct.addCollectionTag(inputTag_);
```

Comment:
No objects are being added to `filterproduct`, so why does this need to be a `HLTFilter` ?

Reply:
Changed to standard EDFilter in 31fb30855064b1366f53d5483f7d7402e7c6ddc1